cocaine/detail/secadaptor.py

```diff
@@ -38,54 +50,82 @@ class TVM(object):
     # in less general name formatting rules.
     TYPE = 'TVM'
 
-    def __init__(self, client_id, client_secret, name='tvm'):
+    def __init__(self, service, credentials):
         """TVM
 
-        :param client_id: Integer client identifier.
-        :param client_secret: Client secret.
-        :param name: TVM service name, defaults to 'tvm'.
+        :param service: TVM service.
+        :param credentials: Credentials for token fetching
         """
-        self._client_id = client_id
-        self._client_secret = client_secret
-
-        self._tvm = Service(name)
+        self._tvm = service
+        self._credentials = credentials
 
     @gen.coroutine
     def fetch_token(self):
         """Gains token from secure backend service.
+        Token allows to visit any destination under TVM protection
 
         :return: Token formatted for Cocaine protocol header.
         """
         grant_type = 'client_credentials'
-
         channel = yield self._tvm.ticket_full(
-            self._client_id, self._client_secret, grant_type, {})
+            self._credentials.client_id,
+            self._credentials.client_secret,
+            grant_type,
+            {}
+        )
         ticket = yield channel.rx.get()
 
-        raise gen.Return(self._make_token(ticket))
+        raise gen.Return(_make_token(self.TYPE, ticket))
 
-    def _make_token(self, ticket):
-        return '{} {}'.format(self.TYPE, ticket)
+
+class TVM2(object):
+    """Tokens fetch interface implementation for TVM protocol with version 2.
```

Comment:
It is `TVM2` itself, not `protocol with version 2`

Reply:
Ok, fixed and squashed.